src/either_or_both.rs

```diff
@@ -500,6 +494,7 @@ impl<T> EitherOrBoth<T, T> {
     }
 }
 
+#[allow(clippy::from_over_into)]
```

Comment:
Does anyone know why we're implementing Into instead of From?

Reply:
I don't see a point of not doing it so I suggest you convert it to a `From` implementation in a separate commit.

Reply:
Here's [an issue](https://github.com/rust-itertools/itertools/issues/836) for that.

Now that shouldn't block this PR?

Reply:
I sure want this PR to be merged as soon as possible as you made quite a work here!